### PR TITLE
Fe in notifications

### PR DESCRIPTION
**Pull request by wiroger9595**
### What this PR does (required):
- add notification bell icon to display how much info is not received
- so far I hard code the data on it, make sure it is will display the data

### Screenshots / Videos (front-end
- no notification screen shot
Screen Shot 2022-02-07 at 9 35 48 PM

-have two notification
Screen Shot 2022-02-07 at 9 44 08 PM

### Any information needed to test this feature (required):
- when notification bell chick, isRead will become true, and will not display anymore

### Any issues with the current functionality (optional):
- have to test when notification list display when it click

### CODE REVIEW

File: client/src/components/Navbar/Notification/Notification.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      <ListItemText
+        primary={`${notification.title}`}
+        secondary={
+          <React.Fragment>
+            <Typography sx={{ display: 'inline' }} component="span" variant="body1" color="text.primary">
+              {`${notification.description}`}
+              <Divider />
+              {`${moment(notification.createdOn).format('MM-DD-YYYY')}`}
+            </Typography>
+          </React.Fragment>
```
Comment (MUGABA) [inline client/src/components/Navbar/Notification/Notification.tsx:35]: There is away to group jsx elements with just an open and close tag `<></>`

File: client/src/helpers/APICalls/notifications.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { FetchOptions } from '../../interface/FetchOptions';
+
+const notifications = async () => {
+  const fetchOptions: FetchOptions = {
+    method: 'GET',
+    credentials: 'include',
+  };
+  return await fetch(`notication/all`, fetchOptions)
```
Comment (MUGABA) [inline client/src/helpers/APICalls/notifications.ts:8]: This can be just `notifications`

**Review by MUGABA — COMMENTED** [review]
Hey can you put all the api calls for the notifications in a single file.

File: client/src/helpers/APICalls/getNotifications.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    credentials: 'include',
+  };
+  return await fetch(`/notification/`, fetchOptions)
+    .then((res) => res.json())
+    .catch(() => ({
+      error: { message: 'Unable to connect to server. Please try again' },
+    }));
+}
+
+export async function fetctAllNotications(): Promise<Notifications> {
```
Comment (moffatethan) [inline client/src/helpers/APICalls/getNotifications.ts:17]: `fetchAllNotifications` please be mindful of spelling

File: client/src/helpers/APICalls/getNotifications.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    credentials: 'include',
+  };
+  return await fetch(`/notification/`, fetchOptions)
+    .then((res) => res.json())
+    .catch(() => ({
+      error: { message: 'Unable to connect to server. Please try again' },
+    }));
+}
+
+export async function fetctAllUnreadNotications() {
```
Comment (moffatethan) [inline client/src/helpers/APICalls/getNotifications.ts:29]: `fetchAllUnreadNotifications`

File: server/controllers/notification.js [hunk 5]
```diff
@@ -34,18 +34,29 @@ exports.createNotification = asyncHandler(async (req, res, next) => {
 // @desc set notification to read
 // @access Private
 exports.markNotificationRead = asyncHandler(async (req, res, next) => {
-  const notificationId = req.params.notificationId;
+  const notificationByUserId = req.params.userId;
+
+  console.log(notificationByUserId);
```
Comment (moffatethan) [inline server/controllers/notification.js:39]: Remove this console log 🙂

File: server/controllers/notification.js [hunk 6]
```diff
@@ -34,18 +34,29 @@ exports.createNotification = asyncHandler(async (req, res, next) => {
...
   const { id: userId } = req.user;
 
-  const notification = await NotificationSchema.findById(notificationId);
+  const notification = await NotificationSchema.find({
+    userId: notificationByUserId,
+    isRead: false,
+  });
+
+  console.log("aaa" + notification);
+  console.log(userId);
```
Comment (moffatethan) [inline server/controllers/notification.js:50]: Let's remove this console log as well

File: server/controllers/notification.js [hunk 7]
```diff
@@ -34,18 +34,29 @@ exports.createNotification = asyncHandler(async (req, res, next) => {
...
 
   if (notification) {
-    if (userId !== notification.recieverId.toString()) {
-      res.status(400);
-      throw new Error("Not authorised to read this");
-    }
+    // if (userId !== notification[0].recieverId) {
+    //   res.status(400);
+    //   throw new Error("Not authorised to read this");
+    // }
```
Comment (moffatethan) [inline server/controllers/notification.js:57]: Let's remove this commented code 🙂

File: server/controllers/notification.js [hunk 8]
```diff
@@ -83,10 +94,16 @@ exports.fetctAllNotications = asyncHandler(async (req, res, next) => {
 exports.fetctAllUnreadNotications = asyncHandler(async (req, res, next) => {
   const { id: userId } = req.user;
 
+  console.log(userId);
+  // console.log(id);
```
Comment (moffatethan) [inline server/controllers/notification.js:98]: Same here please 🙂

File: server/controllers/notification.js [hunk 9]
```diff
@@ -83,10 +94,16 @@ exports.fetctAllNotications = asyncHandler(async (req, res, next) => {
...
+
   const findAllByLoggedUser = await NotificationSchema.find({
     recieverId: userId,
     isRead: false,
-  }).select(["-__v"]);
+  });
+
+  console.log(findAllByLoggedUser);
```
Comment (moffatethan) [inline server/controllers/notification.js:105]: Remove please

**Review by moffatethan — CHANGES_REQUESTED** [review]

File: client/src/components/Notifications/Notifications.tsx [hunk 10]
```diff
@@ -40,8 +35,8 @@ const Notificaitons = (): JSX.Element => {
 
   useEffect(() => {
     const fetchUnReadData = async () => {
-      const getData = await fetctAllUnreadNotications();
-      total.current = Object.keys(getData.data).length;
+      const getData = await fetchAllUnreadNotications();
```
Comment (moffatethan) [inline client/src/components/Notifications/Notifications.tsx:38]: fetchAllUnreadNoti**fi**cations 

Please correct spelling 🙂

File: client/src/helpers/APICalls/getNotifications.ts [hunk 11]
```diff
@@ -14,7 +14,7 @@ export async function createNotification(): Promise<Notifications> {
     }));
 }
 
-export async function fetctAllNotications(): Promise<Notifications> {
+export async function fetchAllNotications(): Promise<Notifications> {
```
Comment (moffatethan) [inline client/src/helpers/APICalls/getNotifications.ts:17]: fetchAllNoti**fi**cations

Comment (moffatethan) [reply client/src/helpers/APICalls/getNotifications.ts:17]: Please update spelling elsewhere in the code

**Review by moffatethan — COMMENTED** [review]
Still some spelling changes I need to see then I'll approve

**Review by moffatethan — APPROVED** [review]

**Pull request merged**